core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -183,6 +180,27 @@ public RecordCursorFactory generate(QueryModel model, SqlExecutionContext execut
         return generateQuery(model, executionContext, true);
     }
 
+    public RecordCursorFactory generateExplain(QueryModel model, RecordCursorFactory factory, int format) {
+        RecordCursorFactory recordCursorFactory = new RecordCursorFactoryStub(model, factory);
+        return new ExplainPlanFactory(recordCursorFactory, format);
+    }
+
+    public RecordCursorFactory generateExplain(ExplainModel model, SqlExecutionContext executionContext) throws SqlException {
+        ExecutionModel innerModel = model.getInnerExecutionModel();
+        QueryModel queryModel = innerModel.getQueryModel();
+        RecordCursorFactory factory;
+        if (queryModel != null) {
+            factory = generate(queryModel, executionContext);
+            if (innerModel.getModelType() != QUERY) {
```

Comment:
is this a leak? If factory is assigne and model is not QUERY, factory will leak memory

Reply:
No, factory is put into ExplainPlanFactory either directly or wrapped in RecordCursorFactoryStub. 
These two tests trigger innerModel.getModelType() != QUERY && innerModel.getModelType() == QUERY  cases : 
```java
    @Test
    public void testExplainInsertAsSelect() throws Exception {
        assertMemoryLeak(() -> {
            compile("create table a ( l long, d double)");
            assertSql("explain insert into a select x, 1 from long_sequence(10)",
                    "QUERY PLAN\n" +
                            "INSERT table: a\n" +
                            "    VirtualRecord\n" +
                            "      functions: [x,1]\n" +
                            "        long_sequence count: 10\n");
        });
    }


    @Test
    public void testGroupByKeyedOnUnionAll() throws Exception {
        assertPlan("create table a ( i int, d double)",
                "select d, max(i) from (select * from a union all select * from a)",
                "GroupBy vectorized: false\n" +
                        "  keys: [d]\n" +
                        "  values: [max(i)]\n" +
                        "    Union All\n" +
                        "        DataFrame\n" +
                        "            Row forward scan\n" +
                        "            Frame forward scan on: a\n" +
                        "        DataFrame\n" +
                        "            Row forward scan\n" +
                        "            Frame forward scan on: a\n");
    }
```